src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
     };
   }
 
+  useEffect(() => {
+    if (clusteringEnabled) {
+      const index = new Supercluster({
+        radius: clusterDistance,
+        maxZoom: 20
+      });
+
+      const points = locations.map((location, index) => ({
+        type: 'Feature',
+        properties: { cluster: false, id: location.id, index },
+        geometry: {
+          type: 'Point',
+          coordinates: [location.position.longitude, location.position.latitude]
+        }
+      }));
+
+      index.load(points);
+      const bounds = [-180, -85, 180, 85];
+      setClusters(index.getClusters(bounds, zoomLevel));
+    } else {
+      setClusters(
+        locations.map((location, index) => ({
+          type: 'Feature',
+          properties: { cluster: false, id: location.id, index },
+          geometry: {
+            type: 'Point',
+            coordinates: [location.position.longitude, location.position.latitude]
+          }
```

Comment:
this is the same code a few lines before, right? i suggest moving it in a function and use it in both places.

Reply:
true, thank you. will be adjusted in the upcoming commit.